staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/marshal_test.go

```diff
@@ -54,9 +59,165 @@ func TestJSONSchemaPropsOrBoolUnmarshalJSON(t *testing.T) {
 			t.Errorf("Failed to unmarshal input '%v': expected %+v, got %+v", c.input, c.result, result)
 		}
 	}
+
+	t.Run("direct invocation", func(t *testing.T) {
+		testCases := []struct {
+			name    string
+			input   []byte
+			wantErr bool
+		}{
+			{
+				name:    "null",
+				input:   []byte(`null`),
+				wantErr: true,
+			},
+			{
+				name:  "zero len",
+				input: []byte{},
+			},
```

Comment:
Another one to add to the list of behaviors in https://github.com/kubernetes/kubernetes/pull/125786. Passing an empty slice to unmarshal would be a caller bug.